File: pytest-api/helpers/paths.py
"""API paths the E2E suite targets."""

from __future__ import annotations

from typing import Final

API_PREFIX: Final = "/api"

LOGIN_URL: Final = f"{API_PREFIX}/login"
ITEMS_URL: Final = f"{API_PREFIX}/items"
HEALTH_URL: Final = f"{API_PREFIX}/health"
OPENAPI_URL: Final = f"{API_PREFIX}/openapi.json"


def item_url(item_id: str) -> str:
    return f"{ITEMS_URL}/{item_id}"

File: web/src/lib/paths.ts
// Derive paths from Vite's BASE_URL so the same build works both:
//   - at "/" (local dev, Docker, Playwright/Cypress CI)
//   - at "/qa-automation-lab/demo/" (GitHub Pages, where the MSW service
//     worker scope is the same subpath and must intercept these requests)
const RAW_BASE: string =
  typeof import.meta !== "undefined" && import.meta.env?.BASE_URL
    ? import.meta.env.BASE_URL
    : "/";
const BASE = RAW_BASE.endsWith("/") ? RAW_BASE.slice(0, -1) : RAW_BASE;

export const API_PREFIX = `${BASE}/api`;

export const LOGIN_URL = `${API_PREFIX}/login`;
export const ITEMS_URL = `${API_PREFIX}/items`;
export const HEALTH_URL = `${API_PREFIX}/health`;

export const itemUrl = (id: string): string => `${ITEMS_URL}/${id}`;
